prime-router/src/main/kotlin/fhirengine/utils/FHIRBundleHelpers.kt

```diff
@@ -116,6 +118,38 @@ fun Bundle.addProvenanceReference() {
     }
 }
 
+/**
+ * Return true if Bundle contains an ELR in the MessageHeader.
+ *
+ * @return true if has a MesssageHeader that contains an R01 or ORU_R01, otherwise false.
+ */
+fun Bundle.isElr(): Boolean {
+    var isElr = false
+    if (this.type == Bundle.BundleType.MESSAGE && this.entry.isNotEmpty()) {
+        // By rule, the first entry must be a MessageHeader
+        val resource = this.entry[0].resource
+        if (resource is MessageHeader) {
+            val event = resource.event
+            if (event is Coding && ((event.code == "R01") || (event.code == "ORU_R01"))) {
```

Comment:
are there other codes that could be ELR in the future? Throwing those codes into a set to check against would make this a bit more readable.

Reply:
That's unknown at this point.  I suspect not right now, though.

Reply:
Unlikely